src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -51,7 +51,12 @@ public FacebookAuthenticationHandler(HttpClient httpClient)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
         {
-            var endpoint = Options.UserInformationEndpoint + "?access_token=" + UrlEncoder.UrlEncode(tokens.AccessToken);
+            var endpoint = Options.UserInformationEndpoint;
+            var accessToken = "access_token=" + UrlEncoder.UrlEncode(tokens.AccessToken);            
+                        
+            endpoint += endpoint.Contains("?") ? "&" : "?";
```

Comment:
Yeah please use QueryHelpers, the manual query string manipulation stood out when I saw the change as well.